evap/evaluation/tests/test_auth.py

```diff
@@ -30,25 +30,29 @@ def test_login_url_generation(self):
         self.assertEqual(reversed_url, '/key/{}'.format(self.external_user.login_key))
 
     def test_login_url_works(self):
-        self.assertRedirects(self.app.get(reverse("contributor:index")), "/?next=/contributor/")
-
         url_with_key = reverse("evaluation:login_key_authentication", args=[self.external_user.login_key])
-        page = self.app.get(url_with_key).follow().follow()
+        page = self.app.get(url_with_key)
+        self.assertContains(page, 'Login')
+        self.assertContains(page, self.external_user.full_name)
+
+        page = self.app.post(url_with_key).follow().follow()
         self.assertContains(page, 'Logout')
+        self.assertContains(page, self.external_user.full_name)
 
     def test_login_key_valid_only_once(self):
-        page = self.app.get(reverse("evaluation:login_key_authentication", args=[self.external_user.login_key])).follow().follow()
-        self.assertContains(page, self.external_user.full_name)
```

Comment:
I'd keep this here (looks like you moved it to the other test? I guess both tests should have it)